config/management/src/validator_config.rs

```diff
@@ -37,81 +35,145 @@ pub struct ValidatorConfig {
 
 impl ValidatorConfig {
     pub fn execute(self) -> Result<Transaction, Error> {
+        // Fetch the owner key from remote storage using the owner_name and derive an address
+        let owner_account = self.fetch_owner_account()?;
+
+        // Create the validator config script for the validator node
+        let validator_config_script = self.create_validator_config_script(owner_account)?;
+
+        // Create and sign the validator-config transaction
+        let validator_config_tx =
+            self.create_validator_config_transaction(validator_config_script)?;
+
+        // Write validator config to local storage to save for verification later on
         let mut local_storage = self.backends.local.create_storage(LocalStorage)?;
+        local_storage
+            .set(
+                constants::VALIDATOR_CONFIG,
+                Value::Transaction(validator_config_tx.clone()),
+            )
+            .map_err(|e| {
+                Error::LocalStorageWriteError(constants::VALIDATOR_CONFIG, e.to_string())
+            })?;
 
-        // Step 1) Retrieve keys from local storage
+        // Save the owner account in local storage for deployment later on
+        local_storage
+            .set(OWNER_ACCOUNT, Value::String(owner_account.to_string()))
+            .map_err(|e| {
+                Error::LocalStorageWriteError(OWNER_ACCOUNT, e.to_string())
+            })?;
+
+        // Upload the validator config to shared storage
+        match self.backends.remote {
+            None => return Err(Error::RemoteStorageMissing),
```

Comment:
why are we making this mandatory? it is entirely possible that I want to perform a manual rotation and this is the tool to do it, we just have to improve the API to export the transaction for submission.

Reply:
As discussed, we can remove this requirement later on (once we update this tool to support post-genesis operations). For now though, we'll leave it in to be explicit.